manifests/init.pp

```diff
@@ -270,6 +271,15 @@
   validate_absolute_path($include_dir_real)
   validate_re($service_ensure, '^(running|stopped)$',
     "nrpe::service_ensure must be \'running\' or \'stopped\'. Detected value is <${service_ensure}>.")
+  if $nrpe_package_provider != undef {
+    validate_string($nrpe_package_provider)
+  }
+
+  if $nrpe_package_provider != undef {
+    Package {
+      provider => $nrpe_package_provider,
+    }
```

Comment:
Had not checked this suggestion. Will move the Package just below the validation line.
Thanks,